openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -116,6 +144,15 @@ def can_load_courseware(self):
             check_if_authenticated=True,
         ).to_json()
 
+        # Only check whether the MFE is enabled if the user would otherwise be allowed to see it
+        # This means that if the user was denied access, they'll see a meaningful message first if
+        # there is one.
+        if access_response and not self.is_microfrontend_enabled_for_user():
```

Comment:
Should we check if the feature is enabled before we check if they have access?

At the end of the day, it shouldn't matter, but if the `access` check is a "heavier" check, we could spare ourselves the access lookups `iff` the feature is toggled off...

I believe the `enabled` check is lighter (no lookups, it should just be checking settings/properties on the arguments).

Reply:
I also believe this would simplify the logic in this function a bit.

Reply:
Actually, on second thought, I don't think this actually buys us anything...
- if the MFE is enabled, we just return the value of the lookup
- if the MFE is disabled, we still need to check `access_response`

The only way it seems worth the change is if we were to change the logic to kick them over to the MFE and rely on _it_ to kick them back, which goes against the comment provided here.

So unless I'm misreading this _again_, I think this is good as-is :)